File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/util/AppContext.java
package de.dimedis.mobileentry.util;

import android.content.Context;
import android.content.pm.PackageManager;
import android.util.Log;

import androidx.annotation.NonNull;

public class AppContext {
    static final String TAG = "AppContext";
    private static Context sContext;
    public static int hasBBApi = -1;

    private AppContext() {
    }

    @NonNull
    public static Context get() {
        return sContext;
    }

    public static void init(@NonNull Context context) {
        sContext = context.getApplicationContext();
    }

    public static boolean hasBBApi() {
        hasBBApi = 1;
        return true;
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/util/AppContext.kt
package de.dimedis.mobileentry.util

import android.content.Context
import android.content.pm.PackageManager
import android.util.Log

object AppContext {
    val TAG = "AppContext"
    lateinit var sContext: Context
    var hasBBApi = -1
    fun get(): Context {
        return sContext!!
    }

    fun init(context: Context) {
        sContext = context.applicationContext
    }

    fun hasBBApi(): Boolean {
        hasBBApi = 1
        // if value already read, do not touch package manager
        if (hasBBApi != -1) {
            return hasBBApi > 0
        }
        val pm = get().packageManager
        try {
            Log.d(TAG, "package: " + pm.getPackageInfo("kr.co.bluebird.android.bbapi.barcodelibconnector",
                    PackageManager.GET_ACTIVITIES))
            hasBBApi = 1
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
            hasBBApi = 0
        }
        return hasBBApi > 0
    }
}